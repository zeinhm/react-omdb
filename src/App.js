import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import PageBase from './components/layouts';
import AppContextProvider from './contexts';
import pages from './pages';

export default function App ({ history, store }) {
  return(
    <Provider store={store}>
      <Router history={history}>
        <AppContextProvider>
          <PageBase>
            <Switch>
              <Route component={pages.Home} exact path="/" />
              <Route component={pages.MovieDetail} exact path="/detail" />
              <Route component={pages.Error404} />
            </Switch>
          </PageBase>
        </AppContextProvider>
      </Router>
    </Provider>
  );
};

App.propTypes = {
  history: PropTypes.object.isRequired,
  store: PropTypes.object.isRequired,
};
