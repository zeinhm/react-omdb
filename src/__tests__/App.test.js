// import React from 'react';
// import ReactDOM from 'react-dom';
// // import store from '../store';

// const mockComponent = (name, props) => <div className={name} {...props} />;

// jest.mock('../App', () => ({
//   __esModule: true,
//   default: props => mockComponent('App', props),
// }));
// jest.mock('../store', () => ({
//   __esModule: true,
//   default: jest.fn(() => ({})),
//   history: {},
// }));
// jest.mock('react-dom');

// describe('src/index', () => {
//   test('render', () => {
//     // expect(store).toHaveBeenCalledTimes(1);
//     expect(ReactDOM.render).toHaveBeenCalledTimes(0);
//   });
// });

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../App';

describe('src/App', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(),
      subscribe: jest.fn(),
    };
    const tree = shallow.render(<App history={{}} store={store} />);
    expect(tree).toMatchSnapshot();
  });
});

