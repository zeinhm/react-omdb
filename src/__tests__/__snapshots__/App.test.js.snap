// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/App render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <BrowserRouter
    history={Object {}}
  >
    <AppContextProvider>
      <PageBase>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/detail"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </PageBase>
    </AppContextProvider>
  </BrowserRouter>
</Provider>
`;
